pkg/mongoproxy/plugins/schema/util.go

```diff
@@ -141,6 +142,29 @@ func Mapify(d bson.D) bson.M {
 	return m
 }
 
+// Map creates a map from the elements of the D with operator
+// It makes additional process for arrays
+func MapifyWithOp(d bson.D, m bson.M) bson.M {
+	for _, e := range d {
+		e := processArray(e)
+		itemType := fmt.Sprint(reflect.TypeOf(e.Value))
+		if itemType == "primitive.D" {
```

Comment:
let's try not to hardcode in codes. we can probably use golang type assertion (equivalent to isinstancof)

Reply:
fixed